codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2ConnectionHandlerBuilder.java

```diff
@@ -566,7 +566,8 @@ private T buildFromConnection(Http2Connection connection) {
         }
 
         DefaultHttp2ConnectionDecoder decoder = new DefaultHttp2ConnectionDecoder(connection, encoder, reader,
-                promisedRequestVerifier(), isAutoAckSettingsFrame(), isAutoAckPingFrame());
+            promisedRequestVerifier(), isAutoAckSettingsFrame(), isAutoAckPingFrame());
+        decoder.validateHeaders(isValidateHeaders());
```

Comment:
This should be set via the constructor. If there is none yet just add one.

Reply:
I did, my original thoughts updated the constructors to add the flag. Hit issues while building because of an API signature change which is why I made this change. Do I just push it like that with the API check error I get building locally?

Reply:
Just add a new constructor and deprecate the old one.

Reply:
Just updated the commit. Please let me know if this is what you meant